tests/_grpc_utils.py

```diff
@@ -24,11 +24,12 @@ def __init__(self):
             self.server_port = int(first_line.replace(b"Server listening on port ", b"").strip())
 
             self._stdout_thread = threading.Thread(
-                target=self._discard_output, args=(self._proc.stdout,), daemon=True
+                target=self._proc.communicate, args=(), daemon=True
             )
             self._stdout_thread.start()
         except Exception:
             self._proc.kill()
+            _, _ = self._proc.communicate()
```

Comment:
I don't understand this part.

Reply:
Why do you need to call communicate after killing the process?

Reply:
I added a comment explaining it.